tests/test_lua_details.cc

```diff
@@ -0,0 +1,174 @@
+
+#include "../src/lua_details.h"
+
+using namespace std::literals;
+using namespace task;
+using namespace Catch::Matchers;
+
+TEST_CASE("execute_lua_script_safely(): Return values from simple scripts without errors",
+    "[lua_details]")
+{
+    sol::state lua;
+
+    SECTION("Empty script")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<sol::object>(*result) == sol::nil);
+    }
+
+    SECTION("return nil")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return nil");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<sol::object>(*result) == sol::nil);
+    }
+
+    SECTION("return true")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return true");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<bool>(*result) == true);
+    }
+
+    SECTION("return false")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return false");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<bool>(*result) == false);
+    }
+
+    SECTION("return 42")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 42");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<int>(*result) == 42);
+    }
+
+    SECTION("return 4.2")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 4.2");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<double>(*result) == 4.2);
+    }
+
+    SECTION("return 'pippo'")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 'pippo'");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<std::string>(*result) == "pippo");
+    }
+}
+
+TEST_CASE("execute_lua_script_safely(): Lua exceptions", "[lua_details]")
+{
+    sol::state lua;
+    open_safe_library_subset(lua); // for error() and pcall()
+
+    SECTION("Syntax error")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "not a lua program");
+        auto* msg = std::get_if<std::string>(&result_or_error);
+        REQUIRE(msg != nullptr);
+        REQUIRE(*msg != "");
+    }
+
+    SECTION("Runtime error")
+    {
+        auto result_or_error = execute_lua_script_safely(
+            lua, "function boom(); error('pippo', 0); end; boom()");
```

Comment:
```suggestion
            lua, "function boom(); error('pip' .. 'po', 0); end; boom()");
```
Well, not needed here, because you are back to 'starts with' and the complete string in one go.

I separated that into 2 parts to accommodate changes in the concrete report-string, that did happen when we initially changed to Sol3.3 without the other changes.

I believe we should check all these outputs much more, but with regexes that are rather lax to avoid test-refactoring on every change ;)

Reply:
Done.